lib/api_resources/servers.js

```diff
@@ -430,11 +430,38 @@ ServerResource.prototype.deviceAction = function(env, next) {
     var args = [body.action];
 
     if (action.fields && action.fields.length) {
-      action.fields.forEach(function(field) {
+
+      var argsOk = action.fields.every(function(field) {
```

Comment:
@AdamMagaluk Instead of failing on the first incorrect field type, could we process all fields and return a summary of every error in the request?